pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+        Vector attribute in the given direction. Used for determining the
+        search direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+#    bounds_1d = np.array(bounds_1d)
+    if vec_1d > 0:
+        idx = np.searchsorted(bounds_1d, cor, side='right')
+    else:
+        idx = np.searchsorted(bounds_1d, cor, side='left')
+    if idx == 0 or idx == len(bounds_1d):
+    # idx could be -1, that means point outside the mesh
+        return -1
+    else:
+        return idx - 1
+
+def _calc_vec_dir(start, end):
+    """
+    Calculate the direction from start to end (start -> end).
+    
+    Parameters:
+    -----------
+    start : numpy array
+        An array of size 3. The start point.
+    end :
+        An array of size 3. The end point.
+
+    Return:
+    -------
+    vec : numpy array
+        Direction from start to end, an unit vector.
+    """
+    return np.divide(end-start, np.linalg.norm(end-start))
+
+def _find_next_grid_1d(cor, vec_1d, bounds_1d):
+    """
+    Find the next grid coordinate of a direction.
+
+    Parameters:
+    -----------
+    cor : float
+        Coordinate of current point in the direction.
+    vec_1d : float
+        Direction value.
+    bounds_1d : numpy array
+        Boundary values of current direction.
+
+    Return:
+    -------
+    n_grid : float
+        The coordinate value of next mesh grid.
+    """
+    voxel_idx = _find_voxel_idx_1d(bounds_1d, cor, vec_1d)
+    if voxel_idx == -1:
+        return vec_1d * np.inf
+    else:
+        if vec_1d > 0:
+            return bounds_1d[voxel_idx+1]
+        else:
+            return bounds_1d[voxel_idx]
+
+def _calc_max_travel_length_in_current_voxel(point, end, vec, bounds):
+    """
+    Calculate the maximum travel length in current voxel before reach next
+    boundary.
+    
+    Parameters:
+    -----------
+    point : numpy array
+        An array of size 3. Temporary point position.
+    end : numpy array
+        An array of size 3. End point
+    vec : numpy array
+        Direction vector from start to end: [v_x, v_y, v_z].
+    bounds : numpy array
+        Boundaries of the mesh grid.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+
+    Return:
+    -------
+    dist_maxs : numpy array
+        The maximum travel lenght in current voxel for 3 directions. Format: 
+        [dist_max_x, dist_max_y, dist_max_z]. These value could be both
+        positive and negtive. And could be 'inf' or '-inf'.
+    """
+    n_x_grid = _find_next_grid_1d(point[0], vec[0], bounds[0])
+    n_y_grid = _find_next_grid_1d(point[1], vec[1], bounds[1])
+    n_z_grid = _find_next_grid_1d(point[2], vec[2], bounds[2])
+    next_grid = np.array([n_x_grid, n_y_grid, n_z_grid])
+    return np.divide(next_grid - point, vec)
```

Comment:
Why divide by the unit vector "vec"?. If I understand correctly, you calculated the distance from the point to the next grid boundaries and you need to get the components that correspond to a vector from the point to the boundaries along "vec", so I guess you need the projection of "next_grid - point" on "vec!

Also create a variable and add a comment above to describe what you're calculating then return it
```suggestion
    # Calculate .......
    dist_maxs = np.divide(next_grid - point, vec)
    return dist_maxs
```